packages/daf-core/src/identity/key-store.ts

```diff
@@ -8,13 +9,19 @@ import Debug from 'debug'
 const debug = Debug('daf:key-store')
 
 export class KeyStore extends AbstractKeyStore {
-  constructor(private dbConnection: Promise<Connection>) {
+  constructor(private dbConnection: Promise<Connection>, private secretBox?: AbstractSecretBox) {
     super()
+    if (!secretBox) {
+      console.warn('Please provide SecretBox to the KeyStore')
+    }
   }
 
   async get(kid: string) {
     const key = await (await this.dbConnection).getRepository(Key).findOne(kid)
     if (!key) throw Error('Key not found')
+    if (this.secretBox && key.privateKeyHex) {
```

Comment:
I'm still uncomfortable that `Key` objects expose the `privateKeyHex` field.
It can very easily be misused. Even the encrypted form can be used as a private key which is very dangerous since it's not recoverable.